src/sessions.ts

```diff
@@ -374,7 +374,7 @@ export class ClientSession extends TypedEventEmitter<ClientSessionEvents> {
    */
   startTransaction(options?: TransactionOptions): void {
     if (this[kSnapshotEnabled]) {
-      throw new MongoCompatibilityError('Transactions are not allowed with snapshot sessions');
```

Comment:
After fixing `isClientError` this started failing because of [this test](https://github.com/mongodb/specifications/blob/master/source/sessions/tests/unified/snapshot-sessions.yml#L482). This feels low impact to fix here.